gcloud-java-bigquery/src/main/java/com/google/gcloud/bigquery/BigQuery.java

```diff
@@ -591,6 +592,33 @@ public static QueryResultsOption maxWaitTime(long maxWaitTime) {
   InsertAllResponse insertAll(InsertAllRequest request) throws BigQueryException;
 
   /**
+   * Sends a resumable insert request given a seekable channel containing the rows to be inserted.
+   * This method does not close the channel so you should take care of closing it.
+   *
+   * <p>Example usage of inserting data from a local file:
+   * <pre> {@code
+   * LoadConfiguration config = LoadConfiguration.of(TableId.of("my_dataset_id", "my_table_id"));
+   * try(FileChannel channel = FileChannel.open(Paths.get("/path/to/file"))) {
+   *   bigquery.insertAll(config, channel);
+   * }}</pre>
+   *
+   * @throws BigQueryException upon failure
+   * @see <a href="https://cloud.google.com/bigquery/loading-data-post-request#resumable">Resumable
+   *     Upload</a>
+   */
+  void insertAll(LoadConfiguration configuration, SeekableByteChannel channel)
```

Comment:
> Yes that could be done, what discouraged me is this sentence from the docs

Actually, in addition to being consistent, that section mention 2 more reasons to support it.
Running on platforms that limit bytes/time per request and supporting frameworks that run
in chunks/slices (e.g. MR).

> However I understand chunking might be useful so I am happy to change.

We could support both if you want. One similar to `Storage.create` with `InputStream` and
one similar to `Storage.write`.

> If I am not mistaken BufferedInputStream supports mark/reset by buffering all data into a byte array, for large data this might become a problem.

It does not necessarily buffer *all* data but rather a configurable size or when mark/reset is used it will buffer up-to the limit. So, it is true that if we unbound the limit that would not be good but if we write in chunks it could potentially used. 
Regardless, I think that you are right not using it considering current implementation
and using status to get the last position and I would keep it this way if we decide to provide both ways.